src/OpenTelemetry.SemanticConventions/SemanticConventions/v1_23_1_Experimental/HttpMetrics.cs

```diff
@@ -0,0 +1,102 @@
+// <copyright file="HttpMetrics.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+// <auto-generated> This file has been auto generated from buildscripts/semantic-conventions/templates/Metrics.cs.j2</auto-generated>
+
+using System.Diagnostics.Metrics;
+
+namespace OpenTelemetry.SemanticConventions.v1_23_1_Experimental
+{
+  /// <summary>
+  /// Describes semantic conventions for metrics in the <c>http</c> namespace.
+  /// </summary>
+  public static class HttpMetrics {
+      /// <summary>
+      /// Creates <c>http.client.request.body.size</c> instrument.
+      /// Size of HTTP client request bodies.
+      /// </summary>
+      /// <remarks>
+      /// The size of the request payload body in bytes. This is the number of bytes transferred excluding headers and is often, but not always, present as the <a href="https://www.rfc-editor.org/rfc/rfc9110.html#field.content-length">Content-Length</a> header. For requests using transport encoding, this should be the compressed size.
+      /// </remarks>
+      public static Histogram<double> CreateHttpClientRequestBodySize(Meter meter)
```

Comment:
value type is not part of yaml - https://github.com/open-telemetry/semantic-conventions/issues/591, so it's not fully possible yet